executors/cmdexec/snapshot.go

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package cmdexec
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/heketi/heketi/executors"
+	"github.com/lpabon/godbc"
+)
+
+func (s *CmdExecutor) snapshotActivate(host string, snapshot string) error {
+	godbc.Require(host != "")
+	godbc.Require(snapshot != "")
+
+	type CliOutput struct {
+		OpRet        int                    `xml:"opRet"`
+		OpErrno      int                    `xml:"opErrno"`
+		OpErrStr     string                 `xml:"opErrstr"`
+		SnapActivate executors.SnapActivate `xml:"snapActivate"`
+	}
+
+	command := []string{
+		fmt.Sprintf("gluster --mode=script --xml snapshot activate %v", snapshot),
+	}
+
+	output, err := s.RemoteExecutor.RemoteCommandExecute(host, command, 10)
+	if err != nil {
+		return fmt.Errorf("Unable to activate snapshot %v: %v", snapshot, err)
+	}
+
+	var snapActivate CliOutput
+	err = xml.Unmarshal([]byte(output[0]), &snapActivate)
+	if err != nil {
+		return fmt.Errorf("Unable to parse output from activate snapshot %v: %v", snapshot, err)
+	}
+	logger.Debug("%+v\n", snapActivate)
+
+	return nil
+}
+
+func (s *CmdExecutor) snapshotDeactivate(host string, snapshot string) error {
+	godbc.Require(host != "")
+	godbc.Require(snapshot != "")
+
+	type CliOutput struct {
+		OpRet          int                      `xml:"opRet"`
+		OpErrno        int                      `xml:"opErrno"`
+		OpErrStr       string                   `xml:"opErrstr"`
+		SnapDeactivate executors.SnapDeactivate `xml:"snapDeactivate"`
+	}
+
+	command := []string{
+		fmt.Sprintf("gluster --mode=script --xml snapshot deactivate %v", snapshot),
+	}
+
+	output, err := s.RemoteExecutor.RemoteCommandExecute(host, command, 10)
+	if err != nil {
+		return fmt.Errorf("Unable to deactivate snapshot %v: %v", snapshot, err)
+	}
+
+	var snapDeactivate CliOutput
+	err = xml.Unmarshal([]byte(output[0]), &snapDeactivate)
+	if err != nil {
+		return fmt.Errorf("Unable to parse output from deactivate snapshot %v: %v", snapshot, err)
+	}
+	logger.Debug("%+v\n", snapDeactivate)
```

Comment:
not checking for opRet here...Command could have been executed but return value could indicate unsuccessful execution.

Reply:
done